src/msgs/data_requests/execute/commit_result.rs

```diff
@@ -10,6 +10,51 @@ pub struct Execute {
     pub proof:      String,
 }
 
+impl SignSelf for Execute {
+    type Extra = u64;
+
+    fn proof(&self) -> Result<Vec<u8>> {
+        Ok(hex::decode(&self.proof)?)
+    }
+
+    fn msg_hash(&self, chain_id: &str, contract_addr: &str, dr_height: Self::Extra) -> Result<Hash> {
+        Ok(hash([
+            "commit_data_result".as_bytes(),
+            self.dr_id.as_bytes(),
+            &dr_height.to_be_bytes(),
+            self.commitment.as_bytes(),
+            chain_id.as_bytes(),
+            contract_addr.as_bytes(),
+        ]))
+    }
+}
+
+impl Execute {
+    pub fn new(
+        dr_id: String,
+        commitment: String,
+        public_key: String,
+        signing_key: &[u8],
```

Comment:
This requires some changes in the overlay that I would like to discuss with @FranklinWaller .

Right now, there is a `SedaChain` struct that exposes a method to sign messages:

```rust
seda_chain.identity_sign(&public_key, &message_hash)?;
```

Reply:
Also that `identity_sign` function is a host exported function for the executor/relayer wasms. It is not a biggie, but still some change to take into account. I sort of liked that the private key never left the struct where is it stored. 😅

Reply:
We discussed that using a Factory pattern as this:

```rust
        let execute_factory = stake::Execute::new(
            public_key.clone(),
            self.memo,
            &seda_chain_config.chain_id,
            &contract_address,
            sequence,
        )?;
        
        let message_hash = execute_factory.get_hash();
        let proof = seda_chain.identity_sign(&public_key, &message_hash)?;
        
        let execute_msg = execute_factory.create_message(proof);
```